src/rtkcmn.c

```diff
@@ -1270,6 +1270,26 @@ extern int str2time(const char *s, int i, int n, gtime_t *t)
     *t=epoch2time(ep);
     return 0;
 }
+/* inversion of time2str (contrary to original str2time)------------------------
+* convert substring in string to gtime_t struct
+* args   : char   *s        I   string ("... yyyy/mm/dd hh:mm:ss ...")
+*          int    i,n       I   substring position and width
+*          gtime_t *t       O   gtime_t struct
+* return : status (0:ok,0>:error)
+*-----------------------------------------------------------------------------*/
+extern int str2time_(const char *s, int i, int n, gtime_t *t)
+{
+    double ep[6];
+    char str[256],*p=str;
+    
+    if (i<0||(int)strlen(s)<i||(int)sizeof(str)-1<i) return -1;
+    for (s+=i;*s&&--n>=0;) *p++=*s++; *p='\0';
```

Comment:
It's a `memcpy`. It might be a good idea to use [strlcpy()](https://linux.die.net/man/3/strlcpy) or memcpy_s.